File: android/app/src/main/kotlin/com/example/map_platform_views/MapConstants.kt
package com.example.map_platform_views

object MapConstants {
    // Channel Names
    const val METHOD_CHANNEL = "native_maps_channel"
    const val VIEW_TYPE = "native_map_view"
    
    // Method Names
    const val ADD_MARKER_METHOD = "addMarker"
    const val MOVE_CAMERA_METHOD = "moveCamera"
    const val CLEAR_MARKERS_METHOD = "clearMarkers"
    
    // Parameter Keys
    const val LATITUDE_KEY = "latitude"
    const val LONGITUDE_KEY = "longitude"
    const val TITLE_KEY = "title"
    const val SNIPPET_KEY = "snippet"
    const val ZOOM_KEY = "zoom"
    const val INITIAL_LAT_KEY = "initialLat"
    const val INITIAL_LNG_KEY = "initialLng"
    const val INITIAL_ZOOM_KEY = "initialZoom"
    
    // Cairo, Egypt - Default Location
    const val CAIRO_LATITUDE = 30.0444
    const val CAIRO_LONGITUDE = 31.2357
    const val DEFAULT_ZOOM = 12f
    
    // Error Codes
    const val ERROR_INVALID_ARGS = "INVALID_ARGS"
    const val ERROR_INVALID_COORDS = "Invalid coordinates"
    const val ERROR_INVALID_PARAMS = "Invalid arguments"
}

File: lib/map_constants.dart
class MapConstants {
  // Channel Names
  static const String methodChannel = 'native_maps_channel';
  static const String viewType = 'native_map_view';
  
  static const String addMarkerMethod = 'addMarker';
  static const String moveCameraMethod = 'moveCamera';
  static const String clearMarkersMethod = 'clearMarkers';
  
  static const String latitudeKey = 'latitude';
  static const String longitudeKey = 'longitude';
  static const String titleKey = 'title';
  static const String snippetKey = 'snippet';
  static const String zoomKey = 'zoom';
  static const String initialLatKey = 'initialLat';
  static const String initialLngKey = 'initialLng';
  static const String initialZoomKey = 'initialZoom';
  
  static const double cairoLatitude = 30.0444;
  static const double cairoLongitude = 31.2357;
  static const double defaultZoom = 12.0;
  
  static const double alexLatitude = 31.2001;
  static const double alexLongitude = 29.9187;
  
  static const String addMarkerError = 'Error adding marker';
  static const String moveCameraError = 'Error moving camera';
  static const String clearMarkersError = 'Error clearing markers';
}